client/packages/common/src/ui/layout/tables/material-react-table/utils.test.ts

```diff
@@ -0,0 +1,112 @@
+import { getGroupedRows } from './utils';
+
+describe('getGroupedRows', () => {
+  const mockT = () => '[multiple]';
+
+  it('should return original data when no groupByField is provided', () => {
+    const data = [
+      { name: 'Item 1', category: 'A' },
+      { name: 'Item 2', category: 'B' },
+    ];
+
+    const result = getGroupedRows(data, undefined, mockT);
+
+    expect(result).toEqual(data);
+  });
+
+  it('should return single row as-is when group contains only one item', () => {
+    const data = [
+      { name: 'Item 1', category: 'A' },
+      { name: 'Item 2', category: 'B' },
+    ];
+
+    const result = getGroupedRows(data, 'category', mockT);
+
+    expect(result).toHaveLength(2);
+    expect(result[0]).toEqual({ name: 'Item 1', category: 'A' });
+    expect(result[1]).toEqual({ name: 'Item 2', category: 'B' });
+  });
+
+  it('should create summary row with subRows when group has [multiple] items', () => {
+    const data = [
+      { name: 'Item 1', category: 'A', price: 10 },
+      { name: 'Item 2', category: 'A', price: 20 },
+    ];
+
+    const result = getGroupedRows(data, 'category', mockT);
+
+    expect(result).toHaveLength(1);
+    expect(result[0]).toEqual({
+      name: '[multiple]',
+      category: 'A', // Same value for all items in group
+      price: '[multiple]',
+      subRows: [
+        { name: 'Item 1', category: 'A', price: 10, isSubRow: true },
+        { name: 'Item 2', category: 'A', price: 20, isSubRow: true },
+      ],
+    });
+  });
+
+  it('should preserve equal values in summary row', () => {
+    const data = [
+      { name: 'Item 1', category: 'A', status: 'active' },
+      { name: 'Item 2', category: 'A', status: 'active' },
+      { name: 'Item 3', category: 'A', status: 'active' },
+    ];
+
+    const result = getGroupedRows(data, 'category', mockT);
+
+    expect(result).toHaveLength(1);
+    expect(result[0]!.status).toBe('active'); // Should preserve equal value
+    expect(result[0]!.category).toBe('A'); // Grouping field should be preserved
+  });
+
+  it('should handle object values correctly', () => {
+    const data = [
+      { category: 'A', details: null },
+      { category: 'A', details: { type: 'small', color: 'red' } },
```

Comment:
had a bug with the grouping when a column is an object type - we were only using the first item in the array to determine if the field is object type... when `null` wasn't an object, the field wasn't treated as object.

So added this test to prove out the change, and other tests as a base for the grouping behaviour, if we need to iterate on it in future!